synapse/storage/databases/main/user_directory.py

```diff
@@ -494,6 +495,30 @@ async def should_include_local_user_in_dir(self, user: str) -> bool:
 
         return True
 
+    def _filter_local_users_for_dir_txn(
+        self, txn: LoggingTransaction, users: Collection[str]
+    ) -> Collection[str]:
```

Comment:
Would it make sense to just replace `should_include_local_user_in_dir` (or have it call this?)

Reply:
That was my original plan, but it turns out that the functions called in `should_include_local_user_in_dir` are cached. I'm really not sure how much we are relying on those caches, so decided to leave it as is :shrug:

Reply:
Fair enough.